src/napari_matplotlib/histogram.py

```diff
@@ -65,20 +70,25 @@ def draw(self) -> None:
 
         # Important to calculate bins after slicing 3D data, to avoid reading
         # whole cube into memory.
-        bins = np.linspace(np.min(data), np.max(data), 100)
+        if data.dtype.kind == "i":
+            # Make sure integer data types have integer sized bins
+            step = (np.max(data) - np.min(data)) // 100
```

Comment:
```suggestion
            step = 1 + (np.max(data) - np.min(data)) // 100
```
Otherwise for max - min < 100 the step size is 0!

Reply:
oh yes!